plot_data/core.py

```diff
@@ -859,6 +872,8 @@ def __init__(self, primitive_groups: List[PrimitiveGroup],
             if not isinstance(value, PrimitiveGroup):
                 primitive_groups[i] = PrimitiveGroup(primitives=value)
         self.primitive_groups = primitive_groups
+        if sizes is not None and type(sizes[0]) == int:
```

Comment:
Why do we check for `type(sizes[0]) == int`, here ? It feels like we try to prevent user from providing malformed input data. Is there a specific usecase in which it can happen ? 
Input type (`List[Tuple[float, float]`) is pretty clear so I would say that we don't need to check for it to be well provided.
If this is just for safety purpose, we cannot check everything that could be problematic. Here it would just be the user that provided malformed data

On the other hand, checking for None to set a default value could be a good idea if the None case is not supported.

In the case we really want to use this check, please consider using `isinstance(sizes[0], int)` instead